runelite-client/src/main/java/net/runelite/client/plugins/groundmarkers/GroundMarkerPlugin.java

```diff
@@ -265,28 +269,58 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 		if (hotKeyPressed && event.getOption().equals(WALK_HERE))
 		{
 			MenuEntry[] menuEntries = client.getMenuEntries();
-			menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 1);
+			if (!config.clearLocalMarkers())
+			{
+				menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 1);
+
+				//Create 1 new menu entries (one for marking)
+				MenuEntry menuEntry = menuEntries[menuEntries.length - 1] = new MenuEntry();
+
+				//Set options for marking
+				menuEntry.setOption(MARK);
```

Comment:
This is exactly the same as the code at https://github.com/runelite/runelite/pull/5211/files#diff-c65443644d54b889eb2a4afdcaa07550R296 - I think it might help clear things up if this is moved outside of the conditional.